hoprd/hoprd/src/main.rs

```diff
@@ -167,6 +170,18 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         }),
     ));
 
+    // Create the metadata database
+    let db_path: String = join(&[&cfg.hopr.db.data, "db"]).expect("Could not create a db storage path");
+
+    let hoprd_db = Arc::new(RwLock::new(hoprd_db_api::db::HoprdDb::new(db_path.clone()).await));
```

Comment:
Why you'd nee the HOPRd DB to be behind an RwLock? 
Are its methods mutating its state ?

Reply:
They are not. RwLock removed.